lib/datadog/core/configuration/components.rb

```diff
@@ -57,223 +57,6 @@ def build_telemetry(settings)
-              # > either profiler is supported, or profiler is not supported.
-              # > In the problematic case, it looks like in your case check 1 decides that profiler is not
-              # > supported => doesn't load it, and then check 2 decides that it is => assumes it is loaded and tries to
-              # > start it.
-              #
-              # I was never able to validate if this was the issue or why exactly .supported? would change its mind BUT
-              # just in case it happens again, I've left this check which avoids breaking the user's application AND
-              # would instead direct them to report it to us instead, so that we can investigate what's wrong.
-              #
-              # TODO: As of June 2021, most checks in .supported? are related to the google-protobuf gem; so it's
-              # very likely that it was the origin of the issue we saw. Thus, if, as planned we end up moving away from
-              # protobuf OR enough time has passed and no users saw the issue again, we can remove this check altogether.
-              Datadog.logger.error(
-                'Profiling was marked as supported and enabled, but setup task was not loaded properly. ' \
-                'Please report this at https://github.com/DataDog/dd-trace-rb/blob/master/CONTRIBUTING.md#found-a-bug'
-              )
-
-              return
-            end
-
-            # Load extensions needed to support some of the Profiling features
-            Profiling::Tasks::Setup.new.run
-
-            # NOTE: Please update the Initialization section of ProfilingDevelopment.md with any changes to this method
-
-            if settings.profiling.advanced.force_enable_new_profiler
-              print_new_profiler_warnings
-
-              recorder = Datadog::Profiling::StackRecorder.new
-              collector = Datadog::Profiling::Collectors::CpuAndWallTimeWorker.new(
-                recorder: recorder,
-                max_frames: settings.profiling.advanced.max_frames,
-                tracer: tracer,
-                gc_profiling_enabled: should_enable_gc_profiling?(settings)
-              )
-            else
-              trace_identifiers_helper = Profiling::TraceIdentifiers::Helper.new(
-                tracer: tracer,
-                endpoint_collection_enabled: settings.profiling.advanced.endpoint.collection.enabled
-              )
-
-              recorder = build_profiler_old_recorder(settings)
-              collector = build_profiler_oldstack_collector(settings, recorder, trace_identifiers_helper)
-            end
-
-            exporter = build_profiler_exporter(settings, recorder)
-            transport = build_profiler_transport(settings, agent_settings)
-            scheduler = Profiling::Scheduler.new(exporter: exporter, transport: transport)
-
-            Profiling::Profiler.new([collector], scheduler)
-          end
-
           private
 
           def build_tracer_tags(settings)
```

Comment:
Should some of these methods also be moved to `tracing/components.rb`? I think most (all?) are only needed by the code that builds the tracer.

Reply:
It looks like so. Moved.